### DISCUSSION TITLE
Cross cutting features & Speech Recognition

### DISCUSSION THREAD

**Discussion by Sharafudeen** [oqtane/oqtane.framework#2002] [General]
Hi,

Exporting Excel, PDF, etc., is included in Oqtane's Core and cross cutting features without using any third-party dependencies?

Also, I'm planning to integrate Speech Recognition for navigating Oqtane's prebuilt (& other custom) modules such as Page, File Management, Exception log, etc. What is the best library to implement speech recognition? The idea behind is.. this can be very helpful for disabled people.. I've used before Toolbelt but it's core development in JavaScript. Any library available for Blazor in C#?

Thanks & Regards!

**Comment by sbwalker** [comment]
Oqtane has no native capabilities for exporting Excel or PDF as there are no features on the framework which require this functionality. However you could certainly build a custom module for your own purposes.

Speech recognition is not on the roadmap for Oqtane.... and I have never explored this area. I am not aware of any libraries for Blazor - you would need to do some research.

**Reply by Sharafudeen** [reply]
For Speech Recognition, I've used below library which is in Javascript and used Blazor C#. Hope it can helpful to others.
https://github.com/jsakamoto/Toolbelt.Blazor.SpeechRecognition
